Integration tests for hibernate-reactive-oracle

IvanPetkov23 opened the pull request:
Added tests for Hibernate Reactive with Oracle. Fixes: #23820

Commit:
Add IT for Hibernate reactive with oracle reactive client

yrodiere commented on integration-tests/hibernate-reactive-oracle/pom.xml:
```diff
@@ -0,0 +1,253 @@
+    <description>Hibernate Reactive related tests running with the Oracle database</description>
+
+    <properties>
+        <oracle.reactive.url>oracle:thin:@localhost:1520/hibernate_orm_test</oracle.reactive.url>
```
Existing integration tests seem to use the convention `reactive-<db>.url`, so here it would be `reactive-oracle.url`. And all other occurrences should be updated, obviously.

See:

https://github.com/quarkusio/quarkus/blob/ef1d293cbde0ca250c049de50647d6a1e1ae7c0e/integration-tests/hibernate-reactive-mysql/pom.xml#L17

yrodiere replied:
Looks like there's a misunderstanding? You change makes sense too, but I was talking about the property name: `<oracle.reactive.url>` => `<reactive-oracle.url>`.

IvanPetkov23 replied:
Sorry, for the misunderstanding :)

yrodiere commented on integration-tests/pom.xml:
```diff
@@ -215,6 +215,7 @@
                 <module>hibernate-reactive-mariadb</module>
                 <module>hibernate-reactive-mysql</module>
                 <module>hibernate-reactive-mysql-agroal-flyway</module>
+                <module>hibernate-reactive-oracle</module>
```
This makes the IT module part of the build, but in order for native tests to run on CI we also need to update `native-tests.json`.

See:

https://github.com/quarkusio/quarkus/commit/e1f94a99c03315bfee05b78ad016df7a311d13ee#diff-307c72aded80fc768e1f20cfa0bed2320c3fe4bb815a6996db89d986c972d845L47-L49

Don't forget to add 5 minutes to the timeout, otherwise this added test might make the whole build time out :)

yrodiere reviewed (changes requested):
Thanks! Looking great, I just have a few comments. The main one being about CI and native tests.

yrodiere reviewed (changes requested):
Some comments to make the build pass on Windows -- which doesn't have containers and thus cannot handle DB tests.
Essentially these tests need to be disabled by default, like in every other module, and enabled when passing `-Dtest-containers` to Maven.

IvanPetkov23 commented:
I am sorry about the merge commits in the PR, but I am new to open source, so I am still learning :)

yrodiere commented:
> I am sorry about the merge commits in the PR, but I am new to open source, so I am still learning :)

No problem. You may want to learn about rebasing: https://docs.github.com/en/get-started/using-git/about-git-rebase

If you rebase your branch on `main`, you should be able to get rid of merge commits. Typically I use `git rebase -i main`, but I'd recommend you read the GitHub docs by following the link above :)

IvanPetkov23 commented:
It is getting worse, so I converted the PR to draft :)

yrodiere commented:
Do you need me to do the rebase, or are you still trying? :)

IvanPetkov23 commented:
I am still trying, but without success. If you can do it for me, that would be great! :)

yrodiere commented:
On it

yrodiere commented:
Alright, I force-pushed a rebase.

I must admit you got yourself in quite a situation :sweat_smile: Better avoid merge commits from the start, next time :) You can instruct git to rebase by default when pulling, instead of merging: https://coderwall.com/p/tnoiug/rebase-by-default-when-doing-git-pull

FWIW, here's what I did:

```
git switch main
git pull
git switch <your branch>
git rebase -i main
```

I used "s" for every commit to squash everything:

![image](https://github.com/user-attachments/assets/6b9f36fe-e2ae-422c-be5a-d57fd3cc5d0e)

Got a conflict:

```
Auto-merging .github/native-tests.json
CONFLICT (content): Merge conflict in .github/native-tests.json
error: could not apply 43df92f39ea... Fix git issues
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
hint: Disable this message with "git config set advice.mergeConflict false"
Recorded preimage for '.github/native-tests.json'
Could not apply 43df92f39ea... Fix git issues
```

Solved it manually in my IDE, then:

```
git add -A
git rebase --continue
```

Got another conflict:

```
Auto-merging integration-tests/hibernate-reactive-oracle/pom.xml
CONFLICT (add/add): Merge conflict in integration-tests/hibernate-reactive-oracle/pom.xml
error: could not apply 4a8813cfe65... Updated native-tests.json to support hibernate-reactive-oracle module. Refactored pom.xml to pass build on windows.
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
hint: Disable this message with "git config set advice.mergeConflict false"
Recorded preimage for 'integration-tests/hibernate-reactive-oracle/pom.xml'
Could not apply 4a8813cfe65... Updated native-tests.json to support hibernate-reactive-oracle module. Refactored pom.xml to pass build on windows.
```

Solved it manually in my IDE, then:

```
git add -A
git rebase --continue
```

Got another conflict:

```
CONFLICT (content): Merge conflict in .github/native-tests.json
Auto-merging integration-tests/hibernate-reactive-oracle/pom.xml
CONFLICT (content): Merge conflict in integration-tests/hibernate-reactive-oracle/pom.xml
error: could not apply b702df5da95... Updated native-tests.json to support hibernate-reactive-oracle module. Refactored pom.xml to pass build on windows.
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
hint: Disable this message with "git config set advice.mergeConflict false"
Recorded preimage for '.github/native-tests.json'
Recorded preimage for 'integration-tests/hibernate-reactive-oracle/pom.xml'
Could not apply b702df5da95... Updated native-tests.json to support hibernate-reactive-oracle module. Refactored pom.xml to pass build on windows.
```


Solved it manually in my IDE, then:

```
git add -A
git rebase --continue
```

Then rebase ended successfully, so I force-pushed to rewrite the git history of this branch (don't do this anywhere, but for PRs with a messy history, it's reasonable):

```
git push --force
```

yrodiere commented on integration-tests/hibernate-reactive-oracle/src/main/resources/application.properties:
```diff
@@ -0,0 +1,17 @@
+
+# Reactive config
+quarkus.datasource.reactive=true
+quarkus.datasource.reactive.url=${oracle.reactive.url}
```
@IvanPetkov23 I think this should have been updated to `reactive-oracle.url` to match your other changes.

CI is failing because it wasn't.

IvanPetkov23 commented:
Thanks for the help and the patience :)

yrodiere committed:
Align Oracle DB setup in quarkus-it-hibernate-reactive-oracle on other ITs

E.g. on quarkus-reactive-oracle-client-deployment,
quarkus-it-jpa-oracle.

Because the current setup doesn't seem to work.

yrodiere reviewed (approved):
I added a commit to fix the DB setup, because it didn't work. Also rebased on main. 

It works locally, let's see how it goes on CI.

IvanPetkov23 commented:
It is working locally on my machine too.

yrodiere commented:
I believe something leaves the containers running after integration tests run, leading to problems when running later integration tests.
I can't investigate now, but will have a look, most likely next week :/

yrodiere commented:
Looks like the failures were caused by problems on the `main` branch; rebasing today fixed them.

We're good to go, I'll merge. Thanks again @IvanPetkov23!
